### PR TITLE
[5.1][com_finder] check if finder content plugin is  enabled

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #42296 .

### Summary of Changes
correct the test if finder content plugin is enabled

### Testing Instructions
Disable finder plugins, then start indexer

### Actual result BEFORE applying this Pull Request
no message about disbled plugin

### Expected result AFTER applying this Pull Request

message about disbled plugin

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on e8247ed379aef7bbefd438196adfed3629fb7928<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42299">issues.joomla.org/tracker/joomla-cms/42299</a>.</sub>

**Comment by Fedik** [comment]
I am sorry to say, but the fix is wrong.
Checking for `plugins/content/finder` is needed but you removing it here.

What probably needed to fix #42296 is to do an addittional check, whether at least on plugin is enabled in the finder group.

**Comment by alikon** [comment]
no problem, you are right
...i've just forgot about the existence of plugin 'content/finder' too... 😨 

I'll give it a closer look

File: administrator/language/en-GB/com_finder.ini [hunk 1]
```diff
@@ -110,6 +110,7 @@ COM_FINDER_FILTERS_N_ITEMS_UNPUBLISHED="%d filters unpublished."
 COM_FINDER_FILTERS_N_ITEMS_UNPUBLISHED_1="Filter unpublished."
 COM_FINDER_FILTERS_TABLE_CAPTION="Filters"
 COM_FINDER_FILTERS_TOOLBAR_TITLE="Smart Search: Search Filters"
+COM_FINDER_FINDER_PLUGINS="Smart Search Finder Plugins"
```
Comment (Quy) [inline administrator/language/en-GB/com_finder.ini:114]: Is it necessary to have a separate string? See line 150.

Comment (alikon) [reply administrator/language/en-GB/com_finder.ini:114]: yes cause these are Finder plugin not Content plugin

Comment (Quy) [reply administrator/language/en-GB/com_finder.ini:114]: Change line 151 to the following and delete line 113??

`COM_FINDER_INDEX_PLUGIN_FINDER_NOT_ENABLED_LINK="The Smart Search Finder Plugins are disabled. Changes to content will not update the Smart Search index if you do not enable these plugins."`

**Comment by HLeithner** [comment]
I don't think that this should go in 4.4 because it's not a bug.

File: administrator/language/en-GB/com_finder.ini (hunk 1 above)
Comment (Fedik) [reply administrator/language/en-GB/com_finder.ini:114]: I think it is better with 2 strings. 
First one is specificaly for "Content - Finder" plugin, and new one for all "Finder - ..." plugins

Comment (alikon) [reply administrator/language/en-GB/com_finder.ini:114]: same for me

**Comment by alikon** [comment]
switched to 5.1

File: administrator/language/en-GB/com_finder.ini (hunk 1 above)
Comment (Quy) [reply administrator/language/en-GB/com_finder.ini:114]: You misunderstood me. I am suggesting combining line 114 and line 152 into one string.

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on a455afe131a73395cee2424babe379b1268633be<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42299">issues.joomla.org/tracker/joomla-cms/42299</a>.</sub>

**Comment by paternax** [comment]
I have tested this item :white_check_mark: successfully on a455afe131a73395cee2424babe379b1268633be<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42299">issues.joomla.org/tracker/joomla-cms/42299</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42299">issues.joomla.org/tracker/joomla-cms/42299</a>.</sub>

**Comment by richard67** [comment]
Back to pending.

@alikon Could you solve the conflict in file "administrator/components/com_finder/src/Controller/DisplayController.php"?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42299">issues.joomla.org/tracker/joomla-cms/42299</a>.</sub>

**Comment by Fedik** [comment]
@alikon the original code has been changed, and has been moved from controller to the layout, please see #42447.
Please sync your changes with it.

**Comment by alikon** [comment]
should be aligned now

**Comment by ufuk-avcu** [comment]
I have tested this item :white_check_mark: successfully on 273a98d23056bc046e520553a48aac7c724d14bf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42299">issues.joomla.org/tracker/joomla-cms/42299</a>.</sub>

**Comment by nielsnuebel** [comment]
I have tested this item :white_check_mark: successfully on 273a98d23056bc046e520553a48aac7c724d14bf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42299">issues.joomla.org/tracker/joomla-cms/42299</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42299">issues.joomla.org/tracker/joomla-cms/42299</a>.</sub>

**Comment by crimle** [comment]
I have tested this item :red_circle: unsuccessfully on 273a98d23056bc046e520553a48aac7c724d14bf<br /><br />Plugin Extension - Finder disabled
Components > Smart Search > Index > Button [Index]
There is no error message<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42299">issues.joomla.org/tracker/joomla-cms/42299</a>.</sub>

**Comment by richard67** [comment]
> I have tested this item 🔴 unsuccessfully on [273a98d](https://github.com/joomla/joomla-cms/commit/273a98d23056bc046e520553a48aac7c724d14bf)Plugin Extension - Finder disabled Components > Smart Search > Index > Button [Index] There is no error message

@crimle It needs to disable all plugins in the "Finder" group, i.e. all with names like "Finder - ...".

**Comment by dorisdreher** [comment]
I have tested this item :red_circle: unsuccessfully on 273a98d23056bc046e520553a48aac7c724d14bf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42299">issues.joomla.org/tracker/joomla-cms/42299</a>.</sub>

**Comment by richard67** [comment]
> I have tested this item 🔴 unsuccessfully on [273a98d](https://github.com/joomla/joomla-cms/commit/273a98d23056bc046e520553a48aac7c724d14bf)

@dorisdreher Could you report back what has not worked? Maybe you've disabled the wrong plugins? Check also my previous comment.

**Comment by crimle** [comment]
> > I have tested this item 🔴 unsuccessfully on [273a98d](https://github.com/joomla/joomla-cms/commit/273a98d23056bc046e520553a48aac7c724d14bf)Plugin Extension - Finder disabled Components > Smart Search > Index > Button [Index] There is no error message
> 
> @crimle It needs to disable all plugins in the "Finder" group, i.e. all with names like "Finder - ...".
There is one plugin only when searching for «finder»

**Comment by dorisdreher** [comment]
I have tested this item :white_check_mark: successfully on 273a98d23056bc046e520553a48aac7c724d14bf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42299">issues.joomla.org/tracker/joomla-cms/42299</a>.</sub>

**Comment by crimle** [comment]
I have tested this item :white_check_mark: successfully on 273a98d23056bc046e520553a48aac7c724d14bf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42299">issues.joomla.org/tracker/joomla-cms/42299</a>.</sub>

File: administrator/language/en-GB/com_finder.ini [hunk 2]
```diff
@@ -148,6 +149,7 @@ COM_FINDER_INDEX_NO_DATA="No content has been indexed."
 COM_FINDER_INDEX_OPTIMISE_FINISHED="Optimisation finished."
 COM_FINDER_INDEX_PLUGIN_CONTENT_NOT_ENABLED="The Smart Search Content Plugin is disabled. Changes to content will not update the Smart Search index until the Plugin is enabled."
 COM_FINDER_INDEX_PLUGIN_CONTENT_NOT_ENABLED_LINK="The %s is disabled. Changes to content will not update the Smart Search index if you do not enable this plugin."
+COM_FINDER_INDEX_PLUGIN_FINDER_NOT_ENABLED_LINK="The %s are disabled. Changes to content will not update the Smart Search index if you do not enable these plugins."
```
Comment (bembelimen) [inline administrator/language/en-GB/com_finder.ini:152]: Can't we make one (new?) constant to cover both cases (as they have the same text) and deprecate the other two?

File: administrator/components/com_finder/src/View/Index/HtmlView.php [hunk 3]
```diff
@@ -154,6 +163,11 @@ public function display($tpl = null)
             $this->finderPluginId = FinderHelper::getFinderPluginId();
         }
 
+        // Check that the finder plugins are enabled
+        if (!PluginHelper::getPlugin('finder')) {
```
Comment (bembelimen) [inline administrator/components/com_finder/src/View/Index/HtmlView.php:167]: Just a personal opinion: I don't like this "loose" check here, but I'm aware that the ```getPlugin``` method sucks in general (not fault of this PR).

File: administrator/language/en-GB/com_finder.ini (hunk 2 above)
Comment (alikon) [reply administrator/language/en-GB/com_finder.ini:152]: yes we can
but out of scope of this pr

File: administrator/components/com_finder/src/View/Index/HtmlView.php (hunk 3 above)
Comment (alikon) [reply administrator/components/com_finder/src/View/Index/HtmlView.php:167]: i can agree with you, 
but if my search results are not wrong
in core,
we already have more than 100 of this ...

out of scope of this pr

**Comment by brianteeman** [comment]
Why was this closed. Looks correct to me.

File: administrator/components/com_finder/src/View/Index/HtmlView.php (hunk 3 above)
Comment (brianteeman) [reply administrator/components/com_finder/src/View/Index/HtmlView.php:167]: all the other uses are not just for the type of plugin but for a specific plugin eg `PluginHelper::isEnabled('content', 'vote');
` but in this case it is correct to have a loose check as we are specifically checking to see if any finder plugins are enabled

File: administrator/language/en-GB/com_finder.ini (hunk 2 above)
Comment (brianteeman) [reply administrator/language/en-GB/com_finder.ini:152]: this second string should be different. The other string refers to the content finder plugin which updates the index whenever you update content. This string is for the finder plugins without which no search is performed

**Comment by alikon** [comment]
lack of interest, this pr was started on j4.4

**Comment by alikon** [comment]
redone #44632

**Pull request closed without merging**